packages/stable/src/api/userProfiles/profilesApi.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 Cognite AS
+
+import { BaseResourceAPI } from '@cognite/sdk-core';
+import {
+  UserProfileItem,
+  UserIdentifier,
+  UserProfilesSearchRequest,
+  UserProfilesListResponse,
+} from '../../types';
+
+export class ProfilesAPI extends BaseResourceAPI<UserProfileItem> {
+  /**
+   * [Get the user profile of the principal issuing the request](https://api-docs.cognite.com/20230101/tag/User-profiles/paths/~1profiles~1me/get/)
+   *
+   * ```js
+   * const response = await client.profiles.me();
+   * ```
+   */
+  public me = async (): Promise<UserProfileItem> => {
+    const path = this.url('me');
+
+    const response = await this.get<UserProfileItem>(path);
+    return this.addToMapAndReturn(response.data, response);
+  };
+
+  /**
+   * [List all user profiles](https://api-docs.cognite.com/20230101/tag/User-profiles/paths/~1profiles/get)
+   *
+   * ```js
+   * const response = await client.profiles.list({ limit: 1000, nextCursor: 'abc' });
+   * ```
+   */
+  public list = (scope?: {
+    limit?: number;
+    cursor?: string;
+  }): UserProfilesListResponse => {
+    return super.listEndpoint(this.callListEndpointWithGet, scope);
+  };
+
+  /**
+   * [Retrieve one or more user profiles by ID](https://api-docs.cognite.com/20230101/tag/User-profiles/paths/~1profiles~1byids/post)
```

Comment:
I dont read js, but just a heads up that byids doesn't return in order. Here's how the Python SDK manually sorts results before returning:
https://github.com/cognitedata/cognite-sdk-python/blob/f4c00fe6fcae3fc9d821e3e9ca5444ac0e11d9a9/cognite/client/_api/user_profiles.py#L81

Reply:
Thanks for the heads up! Should probably be updated at some point, but I'll leave this as is since it's how we're currently treating all the retrieve methods.